src/runtime.js

```diff
@@ -611,10 +627,15 @@ class MultidocManager {
    * @private
    */
   attachShadowDoc_(hostElement, url, initParams, builder) {
+    const isCriticalPath = initParams && initParams.isCriticalPath;
```

Comment:
Overall, I think attaching shadow doc is different enough that it warrants a separate `attachShadowShell`.